### PR TITLE
ENG-1930 - feat: add user agent headers to all client requests

### PR DESCRIPTION
**Pull request by lcschy**
## Description

- Adds `User-Agent` and `BT-CLIENT-USER-AGENT` to requests

## Testing required outside of automated testing?

- [x] Not Applicable

### Screenshots (if appropriate):

- [x] Not Applicable

## Rollback / Rollforward Procedure

- [x] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: src/BasisTheory.net/Common/Utilities/UserAgentUtility.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+using static System.Runtime.InteropServices.RuntimeInformation;
+
+namespace BasisTheory.net.Common.Utilities
+{
+    public static class UserAgentUtility
+    {
+        static UserAgentUtility()
+        {
+            Client = "BasisTheory.net";
+            ClientVersion = Assembly.GetAssembly(typeof(UserAgentUtility)).GetName().Version.ToString();
```
Comment (koalaty-code) [inline src/BasisTheory.net/Common/Utilities/UserAgentUtility.cs:15]: `Version.ToSTring()` by default returns 4 segments. Do we want to override to use `.ToString(3)` to only return `<major>.<minor>.<patch>` versions?

File: src/BasisTheory.net/Common/BaseClient.cs [hunk 2]
```diff
@@ -170,6 +175,14 @@ private void SetRequestHeaders(HttpRequestMessage message, RequestOptions reques
...
             if (!string.IsNullOrEmpty(requestOptions?.CorrelationId))
                 message.Headers.Add("BT-TRACE-ID", requestOptions.CorrelationId);
+            
+            var btClientUserAgentString = UserAgentUtility.BuildBtClientUserAgentString(AppInfo);
+            if (!string.IsNullOrEmpty(btClientUserAgentString))
+                message.Headers.Add("BT-CLIENT-USER-AGENT", btClientUserAgentString);
+
+            var userAgentString = UserAgentUtility.BuildUserAgentString(AppInfo);
+            if (!string.IsNullOrEmpty((userAgentString)))
+                message.Headers.Add("User-Agent", userAgentString);
```
Comment (koalaty-code) [inline src/BasisTheory.net/Common/BaseClient.cs:185]: These headers don't change on a per-request basis. Would it be better to set the DefaultRequestHeaders on the HttpClient that gets passed in or after we build the default client? That would save the overhead of computing the user agent strings per request.

File: src/BasisTheory.net/Common/Utilities/UserAgentUtility.cs (hunk 1 above)
Comment (koalaty-code) [reply src/BasisTheory.net/Common/Utilities/UserAgentUtility.cs:15]: How are we dynamically setting the assembly version when we publish the Nuget package? Pulling the assembly version in an existing version of BasisTheory.net always returns `1.0.0.0`.

**Review by koalaty-code — CHANGES_REQUESTED** [review]

File: src/BasisTheory.net/Common/BaseClient.cs (hunk 2 above)
Comment (lcschy) [reply src/BasisTheory.net/Common/BaseClient.cs:185]: The only thing is that you can pass information about your application when creating the different clients (Application/Tokens/etc). Would have to think of a good way to do that with the other approach...

**Review by koalaty-code — APPROVED** [review]

**Pull request merged**